### ISSUE TITLE
Bug: Copy sequence as HTML is only partial HTML

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#273]
Create sequence, Cmd-A to select it all, right click context menu Copy as HTML.
Although the resulting HTML will work in some browsers, we're really missing the minimal required elements.

Best to go for valid HTML5, at this point, I think:

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>title</title>
  </head>
  <body>
         ....current sequence HTML...
  </body>
</html>
```

**Comment by mpogue2** [comment]
Cmd-A doesn't work, you have to select the first item, and shift select the last item...

**Comment by danlyke** [comment]
I had been copy and pasting this into other documents, but I think
you're right, I should put the boilerplate up there.

**Comment by mpogue2** [comment]
There's already a Save SD Sequence As... menu item.  Perhaps we should also allow saving as .html (rather than just .txt)?  I know this is now two suggestions, rather than one, but that HTML/SVG output for sequences is pretty sweet, if I must say... :-)

**Comment by danlyke** [comment]
I'm good with making it a menu item. I'm still trying to figure out just how to use it, 'cause I think that sometimes I want to copy and paste into a working document. Maybe it belongs in both places.

And I need to fix the "Save SD Sequence As...". I also need to work out the correct "enter text, update the SD table" state machine to get redo working.

**Comment by danlyke** [comment]
Okay, I slapped the boilerplate into the copy as HTML. I see the "actionSave_SD_Sequence_As" in mainwindow.ui, but I don't see it attached to anything, like a menu...

**Comment by mpogue2** [comment]
The save code is at line 8488, in SaveSequenceAs().  Right now it's .txt only.  It would be nice to have it also allow .html....

I'll open a second Issue for the corresponding Print action!

**Comment by danlyke** [comment]
Oh, whoops, one thing I just noticed:

I'd meant for the copy to copy text if any cell is selected, but copy
the formation diagram only if the diagram is selected.

Turns out I had that disabled and I was copying both the text and the
diagram if either is selected.

I don't know if you have a preference. I thought it was kinda cool to
be able to only show the given formations, but I'll leave it as-is,
including all formations, for now.
